src/main.rs

```diff
@@ -1,103 +1,102 @@
-use std::{net::SocketAddr, path::Path, sync::Arc, time::Duration};
+use std::{net::SocketAddr, path::Path, time::Duration};
+
+use clap::Parser;
+use futures::{future::TryFutureExt, FutureExt, StreamExt};
+use miette::{miette, IntoDiagnostic, Result};
+use tokio::{spawn, task::JoinHandle, time::interval};
+use tonic::transport::Server;
+use tracing_subscriber::{filter as tracing_filter, layer::SubscriberExt};
 
 mod cli;
-mod gen;
-mod jsonrpc;
+mod proto;
+mod rpc_client;
 mod server;
 mod types;
 mod validator;
 mod wallet;
+mod zmq;
 
-use clap::Parser;
-
-use futures::{future::try_join_all, TryFutureExt};
-use gen::validator::validator_service_server::ValidatorServiceServer;
-use miette::{miette, Result};
-
+use proto::validator::Server as ValidatorServiceServer;
 use server::Validator;
-
-use tonic::transport::Server;
 use wallet::Wallet;
 
-async fn run_server(bip300: &Validator, addr: SocketAddr) -> Result<()> {
-    log::info!("Listening for gRPC on {addr}");
+// Configure logger.
+fn set_tracing_subscriber(log_level: tracing::Level) -> miette::Result<()> {
+    let targets_filter = tracing_filter::EnvFilter::builder()
+        .with_default_directive(tracing_filter::LevelFilter::from_level(log_level).into())
+        .from_env()
+        .into_diagnostic()?;
+    let stdout_layer = tracing_subscriber::fmt::layer()
+        .compact()
+        .with_line_number(true);
+    let tracing_subscriber = tracing_subscriber::registry()
+        .with(targets_filter)
+        .with(stdout_layer);
+    tracing::subscriber::set_global_default(tracing_subscriber)
+        .into_diagnostic()
+        .map_err(|err| miette::miette!("setting default subscriber failed: {err:#}"))
+}
+
+async fn run_server(validator: Validator, addr: SocketAddr) -> Result<()> {
+    tracing::info!("Listening for gRPC on {addr}");
```

Comment:
This prints out `2024-10-07T17:16:38.121426Z  INFO bip300301_enforcer:41: Listening for gRPC on 127.0.0.1:50051` for me. Would it be possible to get it to print `src/main.rs:41` instead?

Reply:
Done!